### PR TITLE
Refactorings to improve portal creation and config

### PR DESCRIPTION
**Pull request by jameshiebert**
This PR make a bunch of changes to the code to help remove redundancy and make configuration smoother.

- [x] Removes *all* of the auth code. It hasn't been used for several years, so I just pulled it out to prevent it from continuing to muddy the waters.
- [x] DRYs up portal creation. Removes a ton of repetitive code in the ``portals/*.py modules``. Now to create a portal, you essentially only have to define three constants and write one short method to list ensemble members.
- [x] Ports all of the config options to environment variables. This enables the removal of a bunch of complicated templating and scripting inside our Docker deployment, in favor of a simple .env file.
- [x] Separates frontends from the backends so that each can be started independently
- [x] Ensures that there's a single WSGI object that you can pass to Flask (or gunicorn or whatever)

### CODE REVIEW

File: pdp/config.env [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+DATA_ROOT=http://tools.pacificclimate.org/dataportal/data
+TITLE=
+ENSEMBLE_NAME=
+# add password for production
+DSN=postgresql://user:pass@host/database
+# add password for production
+PCDS_DSN=postgresql://user:pass@host/database
+JS_MIN=false
+# change for development
+GEOSERVER=URLhttp://tools.pacificclimate.org/geoserver/
```
Comment (corviday) [inline pdp/config.env]: Typo?

File: README.md [hunk 2]
```diff
@@ -66,11 +66,11 @@ pyenv/bin/python setup.py install
 
 ## Configuration
 
-A sample config file is stored in `pdp/config.yaml`. Copy this file to a new location and set an environment variable `PDP_CONFIG` to point to it.
+Configuration of the PDP is accomplished through a set of environment variables. A sample environment file is stored in `pdp/config.env`. This environment file can be sourced in before you run the pdp, included in a Docker deployment or used in any other flexible way.
 
 ```bash
-cp pdp/config.yaml ~/pdp_config.yaml
```
Comment (corviday) [inline README.md]: Do lines 187 and 201 (pdp.yaml files for supervisord) also need to be updated?

**Comment by corviday** [comment]
Perhaps I have set this up incorrectly, but when I run it, the controls do not load.

pdprefactor

Possibly related: the following error message: `TypeError: pdp.init_login is not a function`. Incomplete removal of auth code, perhaps.

File: pdp/config.env (hunk 1 above)
Comment (jameshiebert) [reply pdp/config.env]: I'm going to blame emacs... must have hit some "swap words" macro accidentally.

File: README.md (hunk 2 above)
Comment (jameshiebert) [reply README.md]: Yeah, we don't really use supervisord anymore, but I can make that more clear.

**Comment by jameshiebert** [comment]
OK, thanks for the detailed review! I've pushed commits that I'm fairly confident that fix all of the problems you found. It's *possible* that there are others, so I'll try to put up a test instance to play with.

**Comment by corviday** [comment]
There is a bug that affects selecting an area to download on raster maps.

**To reproduce:**

1. Go to any raster map, such as this one: http://docker-dev02.pcic.uvic.ca:30555/gridded_observations/map/
1. Use the rectangle tool to select an area to download on the map
1. Mouse over the "Download" link. The URL will be something like http://docker-dev02.pcic.uvic.ca:30556/gridded_observations/pr_forcings_new_symap_BCandCol_1950-2006_11DEC2014.nc.nc?pr[0:20819][][]&

**Expected behaviour:** the URL should be something like http://docker-dev02.pcic.uvic.ca:30556/gridded_observations/pr_forcings_new_symap_BCandCol_1950-2006_11DEC2014.nc.nc?pr[0:20819][146:164][217:237]& , with the last two slices populated.

Possibly related error message:
```
XML Parsing Error: mismatched tag. Expected: </link>.
Location: http://docker-dev02.pcic.uvic.ca:30555/gridded_observations/map/
Line Number 46, Column 5:
```

which appears when a rectangle is drawn.

**Comment by jameshiebert** [comment]
Yes! The XML error **was** related. More ncwms_url list problems, which resulted in metadata calls (for the spatial coordinates of rectangles) getting sent to the wrong URLs. Good catch.

**Comment by corviday** [comment]
I'm not sure I have the latest version (dockerhub said docker-dev02 was already up to date when I tried to pull the new version), but the image currently running on docker-dev02 is using the wrong catalog URL in the Downscaled GCMS portal. It should be using:

http://docker-dev02.pcic.uvic.ca:30555/downscaled_gcms/catalog/catalog.json

but instead it is using:

http://docker-dev02.pcic.uvic.ca:30555/catalog/catalog.json

which results in the dataset selection menu not loading for this portal.


The hydrological model portal is having the same issue, using this url:

http://docker-dev02.pcic.uvic.ca:30555/catalog/catalog.json

instead of:

http://docker-dev02.pcic.uvic.ca:30555/hydro_model_out/catalog/catalog.json

**Comment by jameshiebert** [comment]
Yeah, I redployed on docker-dev02 last night, so it is the most up-to-date.

> but the image currently running on docker-dev02 is using the wrong catalog URL in the Downscaled GCMS portal.

Strange... I'm not seeing the same thing...
catalog

**Comment by corviday** [comment]
Not a bug, but someone seems to have moved or deleted some of the datafiles.
```
Unable to open file /storage/data/climate/hydrology/vic/gen1/5var_day_CCSM3_A1B_run1_19500101-20991231.nc: Unable to open file (unable to open file: name = '/storage/data/climate/hydrology/vic/gen1/5var_day_CCSM3_A1B_run1_19500101-20991231.nc', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)
```

**Comment by corviday** [comment]
Looks good!

I am having trouble downloading data; things just time out. Are you able to download http://docker-dev02.pcic.uvic.ca:30556/downscaled_gcms/pr+tasmax+tasmin_day_BCSD+ANUSPLIN300+CanESM2_historical+rcp26_r1i1p1_19500101-21001231.nc.nc?tasmax[54817:55152][246:266][158:176]& or http://docker-dev02.pcic.uvic.ca:30556/gridded_observations/PNWNAmet_wind.nc.nc?wind[24501:24837][377:393][613:654]& ? Is it just my system?

**Comment by jameshiebert** [comment]
> someone seems to have moved.. some of the datafiles.

Yes... HI seems to do that every once in a while. We really need to `ro` gluster data that has already been indexed. In any case, if you see no other problems are we good to merge?

**Comment by corviday** [comment]
LGTM.

**Pull request merged**